pgspecial/dbcommands.py

```diff
@@ -110,17 +105,17 @@ def list_roles(cur, pattern, verbose):
             FROM pg_catalog.pg_user u
             """
 
-    params = []
+    params = {}
     if pattern:
         _, schema = sql_name_pattern(pattern)
-        sql += "WHERE r.rolname ~ %s"
-        params.append(schema)
-    sql = cur.mogrify(sql + " ORDER BY 1", params)
+        sql += "WHERE r.rolname ~ %(rolname)s"
+        params["rolname"] = schema
+    sql += " ORDER BY 1"
 
-    log.debug(sql)
-    cur.execute(sql)
+    log.debug(f"{sql}, {params}")
```

Comment:
Probably with an negligible impact in this context, but usually you don't pass f-string to log, but rather use the "lazy" good-n-old printf-like idiom:
```
$ python3 -m timeit -s "import logging; x = ['foo']*100" "logging.debug(f'x = {x}')"
50000 loops, best of 5: 6 usec per loop
$ python3 -m timeit -s "import logging; x = ['foo']*100" "logging.debug('x = %s', x)"
500000 loops, best of 5: 583 nsec per loop

```

Reply:
👍  good point.